server/src/main/java/io/crate/statistics/PersistedStatsService.java

```diff
@@ -0,0 +1,256 @@
+        Stats result = get(relationName);
+        return result == null ? defaultValue : result;
+    }
+
+    @Override
+    public void remove(RelationName relationName) {
+        try {
+            try (Writer writer = createWriter(dataPath)) {
+                writer.delete(relationName);
+                writer.flush();
+                writer.commit();
+            }
+            cache.invalidate(relationName);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't delete TableStats from disk", e);
+        }
+    }
+
+    @Override
+    public void clear() {
+        try {
+            try (Writer writer = createWriter(dataPath)) {
+                writer.deleteAll();
+                writer.flush();
+                writer.commit();
+            }
+            cache.invalidateAll();
+        } catch (IOException e) {
+            throw new RuntimeException("Can't delete TableStats from disk", e);
+        }
+    }
+
+    @VisibleForTesting
+    @Nullable
+    Stats loadFromDisk(RelationName relationName) {
+        Path path = dataPath.resolve(STATS);
+        if (Files.exists(path)) {
+            try {
+                try (Directory dir = new NIOFSDirectory(path)) {
+                    DirectoryReader reader;
+                    try {
+                        reader = DirectoryReader.open(dir);
+                    } catch (IndexNotFoundException e) {
+                        LOGGER.warn("Try to load Stats from %s but no index exists %s", path.toString(), e.getMessage());
+                        return null;
+                    }
+                    IndexSearcher indexSearcher = new IndexSearcher(reader);
+                    indexSearcher.setQueryCache(null);
+                    Query query = new TermQuery(new Term(RELATION_NAME_FIELD, relationName.fqn()));
+                    Weight weight = indexSearcher.createWeight(query, ScoreMode.COMPLETE_NO_SCORES, 0.0f);
+                    try (IndexReader indexReader = indexSearcher.getIndexReader()) {
+                        for (LeafReaderContext leafReaderContext : indexReader.leaves()) {
+                            Scorer scorer = weight.scorer(leafReaderContext);
+                            if (scorer != null) {
+                                DocIdSetIterator docIdSetIterator = scorer.iterator();
+                                StoredFields storedFields = leafReaderContext.reader().storedFields();
+                                if (docIdSetIterator.nextDoc() != DocIdSetIterator.NO_MORE_DOCS) {
+                                    Document doc = storedFields.document(docIdSetIterator.docID());
+                                    BytesRef binaryValue = doc.getBinaryValue(DATA_FIELD);
+                                    ByteArrayInputStream bis = new ByteArrayInputStream(binaryValue.bytes);
+                                    return new Stats(new InputStreamStreamInput(bis));
```

Comment:
We should probably also introduce a disk format version to be able to make streaming changes later on.

E.g. in `MetadataStateFormat` we use:

```
                var streamOutput = new OutputStreamStreamOutput(new IndexOutputOutputStream(out));
                Version.writeVersion(Version.CURRENT, streamOutput);
                state.writeTo(streamOutput);
```

and:

to read:

```
                                Version version = Version.readVersion(streamInput);
                                streamInput.setVersion(version);
                                return readFrom(streamInput);
```